src/Arachne/Validation/File/ValidatorFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Arachne package.
+ *
+ * (c) Wojtek Gancarczyk <gancarczyk@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Arachne\Validation\File;
+
+use Arachne\Exception;
+use JsonSchema\RefResolver;
+use JsonSchema\Uri\UriRetriever;
+use JsonSchema\Validator;
+
+/**
+ * Class ValidatorFactory
+ * @package Arachne\Validation\Schema
+ * @author Jan Schädlich <schaedlich.jan@gmail.com>
+ */
+class ValidatorFactory
+{
+    /**
+     * @param $fileType
+     * @return JsonFile|XmlFile
+     */
+    public function create($fileType)
+    {
```

Comment:
Right now I'm in a situation where I need to test both data types in one application. I think it is quite common if you work on an application which integrates with different services/webhooks which provide and expect data in different formats. So I thought it would be a good idea to add the file type recognition to the `ArachneContext` class so that it can be used by all steps. 

Of course you are right my implementation broke backwards compatibility. Which is nor fixed :-)